util/abs/provision_pe_xl_nodes.rb

```diff
@@ -85,6 +158,67 @@
 
 PARAMS_JSON
 
+PROVISION_MESSAGE = <<~PROVISION_MESSAGE
+
+  #{PROVISIONING_TXT} pe_xl nodes with the following options:
+    HA: #{HA}
+    Output directory for Bolt inventory and parameter files: #{OUTPUT_DIR}
+    PE version: #{PE_VERSION}
+    AWS EC2 id tag: #{AWS_TAG_ID}
+    AWS EC2 instance type: #{AWS_INSTANCE_TYPE}
+    AWS EC2 volume size: #{AWS_VOLUME_SIZE}
+
+PROVISION_MESSAGE
+
+NOOP_MESSAGE = "*** Running in no-op mode ***"
+NOOP_EXEC = <<~NOOP_EXEC
```

Comment:
Just a note.  The use of the noop mode here seems to be mainly to validate that the options got through.  Cause the output isn't substantially different than the input, and just tells the user it would call some internal methods with some options.  If that is the case... it may have been better as just spec tests...  It's fine, but next time that might be a better way to go if in this kind of case.  If the script does substantial processing of the options, that is when a noop mode makes sense, as a user might want to check that it did what they expected before running.

Reply:
BTW, the most common use of noop modes I have seen is when it will output actual commands it would run on the shell.

Reply:
If it was implemented as spec tests the user wouldn't be able to see the output when trying out the script which was the intent. If this usage does not seem appropriate for `no-op` mode I can remove it.